packages/pillarbox-debug-panel/src/pillarbox-debug-panel.js

```diff
@@ -0,0 +1,221 @@
+ * for displaying a specific video playback metric, optionally with a graphical
+ * representation (using a GraphComponent).
+ */
+class PillarboxDebugPanel extends Component {
+  update_  = () => this.update();
+
+  /**
+   * Creates an instance of a PillarboxDebugPanel.
+   *
+   * @param {import('video.js/dist/types/player.js').default} player - The Video.js player instance.
+   * @param {Object} options - Configuration options for the PillarboxDebugPanel.
+   */
+  constructor(player, options) {
+    super(player, options);
+
+    this.player().one('loadeddata', this.update_);
+    this.player().on('timeupdate', this.update_);
+  }
+
+  dispose() {
+    this.player().off('loadeddata', this.update_);
+    this.player().off('timeupdate', this.update_);
+  }
+
+  /**
+   * Updates all MetricComponent instances within this panel.
+   */
+  update() {
+    this.children().forEach(child => {
+      if (child.update) {
+        child.update();
+      }
+    });
+  }
+}
+
+/**
+ * Helper function to extract an attribute from the VHS tech.
+ *
+ * @param {import('video.js/dist/types/player.js').default} player - The Video.js player instance.
+ * @param {string} attribute - The VHS attribute to extract.
+ *
+ * @returns {string|undefined} The value of the VHS attribute, or undefined if not available.
+ */
+function extractVhsAttribute(player, attribute) {
+  const tech = player.tech({});
+
+  if (!tech?.vhs) return undefined;
+
+  return tech.vhs.playlists.media().attributes[attribute];
+}
+
+/**
+ * Helper function to extract a statistic from the VHS tech.
+ *
+ * @param {import('video.js/dist/types/player.js').default} player - The Video.js player instance.
+ * @param {string} stat - The VHS statistic to extract.
+ *
+ * @returns {number|undefined} The value of the VHS statistic, or undefined if not available.
+ */
+function extractVhsStats(player, stat) {
+  const tech = player.tech({});
+
+  if (!tech?.vhs) return undefined;
+
+  return tech.vhs.stats[stat];
+}
+
+/**
+ * Helper function to get the currently selected video track.
+ *
+ * @param {import('video.js/dist/types/player.js').default} player - The Video.js player instance.
+ *
+ * @returns {Object|undefined} The currently selected video track, or undefined if none is selected.
+ */
+function currentVideoTrack(player) {
+  return Array.from(player.videoTracks()).find(track => track.selected);
+}
+
+/**
+ * Helper function to get the currently enabled audio track.
+ *
+ * @param {import('video.js/dist/types/player.js').default} player - The Video.js player instance.
+ *
+ * @returns {Object|undefined} The currently enabled audio track, or undefined if none is enabled.
+ */
+function currentAudioTrack(player) {
+  return Array.from(player.audioTracks()).find(track => track.enabled);
+}
+
+PillarboxDebugPanel.prototype.options_ = {
+  className: 'pbw-debug-panel',
+  children: [
+    {
+      name: 'MetricComponent',
```

Comment:
Could we give each component its own name to facilitate access to each of them? `player.pillarboxDebugPanel.bufferMetric`
```suggestion
      name: 'bufferMetric',
      classComponent: 'MetricComponent',
```